File: crates/klyron_watcher/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron_watcher {
WatcherBuilder::WatcherBuilder() : config_() {}
WatcherBuilder& WatcherBuilder::enabled(bool v) { config_.enabled = v; return *this; }
WatcherBuilder& WatcherBuilder::verbose(bool v) { config_.verbose = v; return *this; }
WatcherConfig WatcherBuilder::build() { return config_; }
}

File: crates/klyron_watcher/bindings/rust/builder.rs
use crate::config::WatcherConfig;

pub struct WatcherBuilder {
    config: WatcherConfig,
}

impl WatcherBuilder {
    pub fn new() -> Self { Self { config: WatcherConfig::default() } }
    pub fn build(self) -> WatcherConfig { self.config }
}

impl Default for WatcherBuilder {
    fn default() -> Self { Self::new() }
}
